auditlog/models/rule.py

```diff
@@ -0,0 +1,446 @@
+                continue
+            field = self._get_field(log.model_id, field_name)
+            log_vals = self._prepare_log_line_vals_on_create(
+                log, field, new_values)
+            log_line_model.create(log_vals)
+
+    def _prepare_log_line_vals_on_create(self, log, field, new_values):
+        """Prepare the dictionary of values used to create a log line on a
+        'create' operation.
+        """
+        vals = {
+            'field_id': field.id,
+            'field_name': field.name,
+            'field_description': field.field_description,
+            'log_id': log.id,
+            'old_value': False,
+            'old_value_text': False,
+            'new_value': new_values[log.res_id][field.name],
+            'new_value_text': new_values[log.res_id][field.name],
+        }
+        if field.relation and '2many' in field.ttype:
+            new_value_text = self.env[field.relation].browse(
+                vals['new_value']).name_get()
+            vals['new_value_text'] = new_value_text
+        return vals
+
+    @api.multi
+    def subscribe(self):
+        """Subscribe Rule for auditing changes on model and apply shortcut
+        to view logs on that model.
+        """
+        act_window_model = self.env['ir.actions.act_window']
+        model_data_model = self.env['ir.model.data']
+        for rule in self:
+            # Create a shortcut to view logs
+            domain = "[('model_id', '=', %s), ('res_id', '=', active_id)]" % (
+                rule.model_id.id)
+            vals = {
+                'name': _(u"View logs"),
+                'res_model': 'auditlog.log',
+                'src_model': rule.model_id.model,
+                'domain': domain,
+            }
+            act_window = act_window_model.sudo().create(vals)
+            rule.write({'state': 'subscribed', 'action_id': act_window.id})
+            keyword = 'client_action_relate'
+            value = 'ir.actions.act_window,%s' % act_window.id
+            model_data_model.sudo().ir_set(
+                'action', keyword, 'View_log_' + rule.model_id.model,
+                [rule.model_id.model], value, replace=True,
+                isobject=True, xml_id=False)
+        return True
+
+    @api.multi
+    def unsubscribe(self):
+        """Unsubscribe Auditing Rule on model."""
+        act_window_model = self.env['ir.actions.act_window']
+        ir_values_model = self.env['ir.values']
+        value = ''
+        # Revert patched methods
+        self._revert_methods()
+        for rule in self:
+            # Remove the shortcut to view logs
+            act_window = act_window_model.search(
+                [('name', '=', 'View Log'),
+                 ('res_model', '=', 'auditlog.log'),
+                 ('src_model', '=', rule.model_id.model)])
+            if act_window:
+                value = 'ir.actions.act_window,%s' % act_window.id
+                act_window.unlink()
+            if value:
```

Comment:
Strange flow, fixed when you remove this condition and leave the indentation the same. Allows you to remove the declaration of 'value' in line 427 too.